torch/distributed/tensor/_api.py

```diff
@@ -426,6 +426,10 @@ def from_local(
                     if placement.dim < 0:
                         placements[idx] = Shard(placement.dim + local_tensor.ndim)
 
+        # if the backend is xla device, return XLAShardedTensor
+        if device_type == "xla":
+            return xla_distribute_tensor(local_tensor, device_mesh, placements)
```

Comment:
this looks incorrect. `from_local()` has a different semantics than `distribute_tensor()`.